### PR TITLE
feat(workspaces): add shared context

### PR DESCRIPTION
**Pull request by victorarias**
## Intent / Why

Long-running and delegated agents need durable workspace-level context so new sessions can start with current goals, decisions, constraints, and progress instead of reconstructing them from scratch. This introduces the shared context foundation before adding the chief-of-staff role.

## Summary

- add `attn workspace context show`, `update`, and `status` with semi-persistent per-session markdown checkouts
- persist canonical workspace context with revision-based conflict detection and preserve local edits on conflicts
- emit `workspace_context_changed` and keep context-bearing workspaces alive after their last session closes
- expose the operations over socket and WebSocket protocols and bump the protocol version to 91
- teach agents the file-based workflow through the split attn skill reference

## Test plan

- [x] `go test ./...`
- [x] `pnpm --dir app test -- --run` (635 tests)
- [x] `pnpm --dir app run build`
- [x] `pnpm --dir app run e2e` (104 passed, 1 skipped)
- [x] manual dev test: Claude publishes context and Codex reads and updates it in the same workspace
- [x] manual dev test: stale update is rejected while preserving local edits
- [x] manual dev test: context-bearing workspace survives session close and daemon restart, then is removed by explicit workspace deletion

## Out of scope

- injecting or pinging a running agent when another session changes workspace context
- the unique chief-of-staff agent role and tracked dispatch relationships
- Tauri UI for editing or visualizing workspace context

### CODE REVIEW

**Review by figgyster — DISMISSED** [review]
This looks safe to merge. The workspace context path has the right contract shape: revision-checked canonical state, preserved local edits on conflicts, socket/WebSocket coverage, and enough focused tests around the daemon/store/CLI seams.

I ran focused Go coverage for the new context surfaces plus the `useDaemonSocket` suite and TypeScript check locally; visible CI is green too. No blockers from me.

**Review by figgyster — DISMISSED** [review]
Codex caught a real edge case here, and it crosses from polish into user-contract territory.

`UpdateWorkspaceContext` treats the initial empty checkout as a change because the identical-content no-op path only runs when `current.Revision > 0`. So `attn workspace context update` with an untouched empty context creates revision 1, emits a change, and makes `HasWorkspaceContext` keep that workspace alive after the last session closes.

Required before merge:
- Treat `revision=0, content=""` updated with `content=""` as `changed=false` with no row insertion / broadcast / keepalive effect.
- Add a regression test for the untouched initial checkout/update path so this gremlin doesn’t get a lease.

File: internal/store/workspace_context.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	)
+	if errors.Is(err, sql.ErrNoRows) {
+		current = protocol.WorkspaceContext{WorkspaceID: workspaceID}
+	} else if err != nil {
+		return nil, false, fmt.Errorf("read workspace context %s: %w", workspaceID, err)
+	}
+	if current.Revision != expectedRevision {
+		return nil, false, fmt.Errorf("%w: expected revision %d, current revision %d", ErrWorkspaceContextConflict, expectedRevision, current.Revision)
+	}
+	if current.Content == content && current.Revision > 0 {
```
[the remark this answers is not in the document]
Comment (victorarias) [reply internal/store/workspace_context.go]: Fixed in 3d9475b. Identical content is now a no-op at every revision, including the synthetic revision-0 empty state. Regression coverage verifies the untouched initial checkout/update creates no row, queues no workspace_context_changed event, and does not keep the workspace alive after its last session leaves. Full `go test ./...` passes.

**Review by figgyster — APPROVED** [review]
The fix closes the empty-context edge case cleanly. Untouched revision-0 checkouts now stay no-op/no-row/no-keepalive, while real edits still go through the revision-checked update path.

Focused checks passed locally:
- `git diff --check origin/main...HEAD`
- `go test ./internal/store ./internal/daemon ./internal/client ./internal/protocol ./cmd/attn`
- `pnpm --dir app exec vitest run src/hooks/useDaemonSocket.test.tsx`

No blockers from me.
